examples/ex-layout.cpp

```diff
@@ -0,0 +1,197 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
```

Comment:
@davidbeckingsale , @rhornung67  I was wondering if you might be able to checkout this example. I was wondering if it looked good, if so I can clean up the view/layout page around this example.